spark-jobs/src/main/scala/filodb/downsampler/chunk/BatchExporter.scala

```diff
@@ -0,0 +1,223 @@
+  @transient lazy val exportPrepBatchLatency =
+    Kamon.histogram("export-prep-batch-latency", MeasurementUnit.time.milliseconds).withoutTags()
+
+  val rules = downsamplerSettings.exportRules
+  val exportSchema = {
+    val fields = new mutable.ArrayBuffer[StructField](3 + downsamplerSettings.exportPathSpecPairs.size)
+    fields.append(
+      StructField("labels", StringType),
+      StructField("timestamp", LongType),
+      StructField("value", DoubleType)
+    )
+    // append all partitioning columns as strings
+    fields.appendAll(downsamplerSettings.exportPathSpecPairs.map(f => StructField(f._1, StringType)))
+    StructType(fields)
+  }
+  val partitionByCols = {
+    val cols = new mutable.ArrayBuffer[String](3 + downsamplerSettings.exportPathSpecPairs.size)
+    cols.appendAll(downsamplerSettings.exportPathSpecPairs.map(_._1))
+    cols
+  }
+
+  // Unused, but keeping here for convenience if needed later.
+  private def hashToString(bytes: Array[Byte]): String = {
+    MessageDigest.getInstance("SHA-256")
+      .digest(bytes)
+      .map("%02x".format(_))
+      .mkString
+  }
+
+  /**
+   * Returns an iterator of a single column of the argument partition's chunk.
+   * @param iCol the index of the column to iterate.
+   * @param iRowStart the index of the row to begin iteration from.
+   */
+  private def getChunkColIter(partition: ReadablePartition,
+                              chunkReader: ChunkSetInfoReader,
+                              iCol: Int,
+                              iRowStart: Int): TypedIterator = {
+    val tsPtr = chunkReader.vectorAddress(iCol)
+    val tsAcc = chunkReader.vectorAccessor(iCol)
+    partition.chunkReader(iCol, tsAcc, tsPtr).iterate(tsAcc, tsPtr, iRowStart)
+  }
+
+  /**
+   * Extracts and returns an iterator of (timestamp,value) pairs from the argument chunk.
+   * @param irowStart the index of the first row to include in the result.
+   * @param irowEnd the index of the final row to include in the result (exclusive).
+   */
+  private def getTimeValuePairs(partKeyMap: Map[String, String],
+                                partition: ReadablePartition,
+                                chunkReader: ChunkSetInfoReader,
+                                irowStart: Int,
+                                irowEnd: Int): Iterator[(Long, Double)] = {
+    val timestampCol = 0  // FIXME: need a more dynamic (but efficient) solution
+    val columns = partition.schema.data.columns
+    val valueCol = columns.indexWhere(_.name == partition.schema.data.valueColName)
+    val timestampIter = getChunkColIter(partition, chunkReader, timestampCol, irowStart).asLongIt
+    columns(valueCol).columnType match {
+      case DoubleColumn =>
+        val valueIter = getChunkColIter(partition, chunkReader, valueCol, irowStart).asDoubleIt
+        (irowStart until irowEnd).iterator.map{ _ =>
+          (timestampIter.next, valueIter.next)
+        }
+      case colType =>
+        DownsamplerContext.dsLogger.warn(
```

Comment:
I use this `DownsamplerContext` for logging in `BatchExporter`. Should it be something different?